user_tools/src/spark_rapids_tools/configuration/common.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Common types and definitions used by the configurations. This module is used by other
+modules as well."""
+
+from typing import Union
+from pydantic import BaseModel, Field, AnyUrl, FilePath, AliasChoices
+
+from spark_rapids_tools.enums import DependencyType
+from spark_rapids_tools.storagelib.tools.fs_utils import FileHashAlgorithm
+
+
+class RuntimeDependencyType(BaseModel):
+    """Defines the type of dependency. It can be one of the following:
+       - Archived file (.tgz)
+       - Simple JAR file (*.jar)
+       - Classpath directory (not yet supported)
+
+    Note: The 'classpath' type is reserved for future use, allowing users to point to a directory
+    in the classpath without needing to download or copy any binaries."""
+
+    dep_type: DependencyType = Field(
+        description='The type of the dependency',
+        validation_alias=AliasChoices('dep_type', 'depType'))
+    relative_path: str = Field(
+        default=None,
+        description='The relative path of the dependency in the classpath. This is relevant for tar files',
```

Comment:
"This is relevant for tar files".. does this means I can't use .gz files or just .jar files?

Reply:
I will clarify the description. This field is used to tell the tools which subdirectory to find the additional jars. Therefore, it is only relevant with "archive" like spark.tgz because it produces an entire directory.
if the dependency is `.jar` file , then such information is not needed because the jar file is added to the classpath.